src/main/java/chess/dao/RoomDaoImpl.java

```diff
@@ -0,0 +1,80 @@
+package chess.dao;
+
+import chess.dto.RoomDto;
+import java.util.List;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class RoomDaoImpl implements RoomDao {
+
+    private static final String DEFAULT_GAME_STATE = "ready";
+    private static final String FIRST_TURN = "WHITE";
+
+    private final JdbcTemplate jdbcTemplate;
+
+    public RoomDaoImpl(final JdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    private static final RowMapper<RoomDto> actorRowMapper = (resultSet, rowNum) -> new RoomDto(
+            resultSet.getString("name"),
+            resultSet.getInt("roomId")
+    );
+
+
+    @Override
+    public void saveNewRoom(final String roomName, final String password) {
+        final String sql = "insert into room (name, password, gameState, turn) values (?, ?, ?, ?)";
+        jdbcTemplate.update(sql, roomName, password, DEFAULT_GAME_STATE, FIRST_TURN);
+    }
+
+    @Override
+    public boolean hasDuplicatedName(final String roomName) {
+        final String sql = "select count(*) from room where name = ?";
```

Comment:
```suggestion
        final String sql = "select * from room where name = ? LIMIT 1";
```
미션과 별개로 count쿼리를 작성하면, 모든 테이블을 조회하게되는데요. 비효율적일 수 있습니다. 
EXISTS 쿼리나 LIMIT 1을 활용해보는게 좋을 것 같아요.

Reply:
Limit으로 수정했습니다!